setup.py

```diff
@@ -197,13 +197,20 @@ def run(self):
     'testpath', 
 ]
 
-extras_require = setuptools_args['extras_require'] = {
+extra_requirements = {
     # FIXME: tests still require nose for some utility calls,
     # but we are running with pytest
-    'test': ['pytest', 'pytest-cov', 'nose', 'ipykernel', 'jupyter_client'],
+    'test': ['pytest', 'pytest-cov', 'nose', 'ipykernel', 'jupyter_client>=4.2'],
     'serve': ['tornado>=4.0'],
     'execute': ['jupyter_client>=4.2'],
 }
+s = [] 
+[s.extend(values) for values in extra_requirements.values()]
+extra_requirements['all'] = list(set(s))
+
+extras_require = setuptools_args['extras_require'] = {
+    **extra_requirements
```

Comment:
This syntax is new in Python 3.5. If we need a copy of the dictionary, I think `extra_requirements.copy()` should work. I'm not sure that we do need a copy, though - I think you can just set them to the same object.